transport/TransportImpl.cpp

```diff
@@ -2390,44 +2357,27 @@ void TransportImpl::runTick(uint64_t timestamp)
 
 void TransportImpl::doRunTick(const uint64_t timestamp)
 {
-    if (!_config.rctl.enable || !_config.bwe.useUplinkEstimate)
-    {
-        while (!_rtxPacingQueue.empty())
-        {
-            protectAndSendRtp(timestamp, _rtxPacingQueue.fetchBack());
-        }
-        while (!_pacingQueue.empty())
-        {
-            protectAndSendRtp(timestamp, _pacingQueue.fetchBack());
-        }
-        _pacingInUse = false;
-        return;
+    auto drainMode = _config.rctl.enable && _config.bwe.useUplinkEstimate ? UseBudget : DrainAll;
+    drainPacingBuffer(timestamp, drainMode);
+    if (UseBudget == drainMode) {
+        sendPadding(timestamp);
     }
+    _pacingInUse = !_pacingQueue.empty() || !_rtxPacingQueue.empty();
+}
 
-    auto budget = _rateController.getPacingBudget(timestamp);
-    while (!_pacingQueue.empty() || !_rtxPacingQueue.empty())
-    {
-        auto* pacingQueue = &_pacingQueue;
-        if (!_rtxPacingQueue.empty())
-        {
-            pacingQueue = &_rtxPacingQueue;
-        }
+memory::UniquePacket TransportImpl::tryFetchPriorityPacket(size_t budget)
+{
+    auto queue = _rtxPacingQueue.empty() ? _pacingQueue : _rtxPacingQueue;
+    return !queue.empty() && budget >= queue.back()->getLength() + _config.ipOverhead ? queue.fetchBack() : nullptr;
+}
 
-        if (pacingQueue->back()->getLength() + _config.ipOverhead <= budget)
-        {
-            memory::UniquePacket packet(pacingQueue->fetchBack());
-            budget -= packet->getLength() + _config.ipOverhead;
-            protectAndSendRtp(timestamp, std::move(packet));
-        }
-        else
-        {
-            break;
-        }
+void TransportImpl::drainPacingBuffer(uint64_t timestamp, DrainPacingBufferMode mode)
```

Comment:
Instead f adding that enum, why don't we just pass the drain budget to the method. Either SIZE_MAX or _rateController.getPacingBudget. Seems like we add new "types" for no solid reason.